### DISCUSSION TITLE
Implementing multiranger_wall_following on simple_mapper from crazyflie_ros2_experimental

### DISCUSSION THREAD

**Discussion by Arsfake26** [bitcraze/discussions#792] [Q&A] [answered]
Hi,
I'm currently working on a project to make the crazyflie move automatically while mapping. Is it possible to implement wall following code on simple mapper package? Thanks in advance!

**Comment by knmcguire** [comment, marked as answer]
Hi! Ofcourse that would be possible :) but it hasn't been implemented in that combination so that is something you'll need to do yourself.

You can either:
- Make a ROS package that is inspired by the [python version of the wallfollowing algorithm](https://github.com/bitcraze/crazyflie-lib-python/blob/master/examples/multiranger/multiranger_wall_following.py) that has as input the multiranger values as topics and outputs twist velocities.
- Don't control the crazyflie's movement through ROS but use [the onboard wall following example](https://github.com/bitcraze/crazyflie-firmware/tree/master/examples/demos/app_wall_following_demo) but keep logging the multiranger values and positioning such that the simple mapper can use it. 

I do think it is a better idea to use the Crazyswarm2 project for this as it is possible to connect that to the SLAM toolbox (see [these tutorials](https://imrclab.github.io/crazyswarm2/tutorials.html#mapping-with-the-slam-toolbox)) The simple mapper hasn't been implemented there yet but should be easy to do. But just because the crazyflie_ros2_experimental is no longer maintained.
